lib/ui/src/containers/panel.js

```diff
@@ -10,8 +10,25 @@ const createPanelActions = memoize(1)(api => ({
   togglePosition: () => api.togglePanelPosition(),
 }));
 
+const filterPanel = (panels, storyParameters) => {
+  if (!panels || !storyParameters) {
+    return panels;
+  }
+
+  const filteredPanels = {};
+  Object.entries(panels).forEach(([id, panel]) => {
+    const { paramKey } = panel;
+    const panelParameters = paramKey && storyParameters[paramKey];
+    if (!panelParameters || !panelParameters.disabled) {
+      filteredPanels[id] = panel;
+    }
+  });
+
+  return filteredPanels;
+};
+
 const mapper = ({ state, api }) => ({
-  panels: api.getPanels(),
+  panels: filterPanel(api.getPanels(), api.getParameters(state.storyId)),
```

Comment:
should `getFilteredPanels` be an api?
should `getPanels` just return the filtered list

We should make sure this is memoized

Reply:
I added a `getPanelForStory(storyParameters)` in the addon api.
Since the `api.getParameters()` is in the story api, it feels wrong to use it in the addon api. My first thought is that those are separate modules.

Reply:
The modules addons, layout, stories are all interconnected.

It's OK to call methods on the final / complete api in methods of the SubAPI.

It's probably not a best practice or anything, but duplicating code and making work arounds so "modules" **appear** self-contained when in fact they aren't really is maybe even worse?

Reply:
You're right, and I haven't noticed it but the addon api already use other apis.
Changed it :)